src/framework/mlt_producer.h

```diff
@@ -97,6 +97,12 @@ struct mlt_producer_s
 	 */
 	int ( *set_in_and_out )( mlt_producer, mlt_position, mlt_position );
 
+	/** Probe for metadata.
+	 *
+	 * \param mlt_producer a producer
+	 * \return true if there was an error
+	 */
+	int ( *probe )( mlt_producer );
```

Comment:
This breaks ABI, which is not acceptable.

Reply:
Does this mean it is impossible to add a new member to the producer struct? I guess I could store it in a property.

Reply:
It is temping to simply move the new member to the end of struct, and any theoretical existing built code that is allocating the struct and/or copying it cannot benefit from the new feature anyways. However, there are other structs that embed this struct first, and then it changes the offset of its members that follow! Maybe instead of using a property use a mlt_event.